modules/test/playwright/tests/frontend-taglib-clay/main/icons.spec.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {expect, mergeTests} from '@playwright/test';
+
+import {apiHelpersTest} from '../../../fixtures/apiHelpersTest';
+import {featureFlagsTest} from '../../../fixtures/featureFlagsTest';
+import {isolatedSiteTest} from '../../../fixtures/isolatedSiteTest';
+import {loginTest} from '../../../fixtures/loginTest';
+import {clickAndExpectToBeVisible} from '../../../utils/clickAndExpectToBeVisible';
+import getRandomString from '../../../utils/getRandomString';
+import getPageDefinition from '../../layout-content-page-editor-web/main/utils/getPageDefinition';
+import getWidgetDefinition from '../../layout-content-page-editor-web/main/utils/getWidgetDefinition';
+
+const EXPECTED_FLAG_CODES = [
+	'en-us',
+	'ar-sa',
+	'ca-es',
+	'zh-cn',
+	'nl-nl',
+	'fi-fi',
+	'fr-fr',
+	'de-de',
+	'hu-hu',
+	'ja-jp',
+	'pt-br',
+	'es-es',
+	'sv-se',
+];
+
+const test = mergeTests(
+	apiHelpersTest,
+	isolatedSiteTest,
+	loginTest(),
+	featureFlagsTest({
+		'LPS-178052': {enabled: true},
+	})
+);
+
+test(
+	'Verify that flag icons are present when clicking on localization button',
+	{tag: '@LPD-66978'},
+	async ({apiHelpers, page, site}) => {
+		await test.step('Create a page with Language Selector widget', async () => {
+			const widgetDefinition = getWidgetDefinition({
+				id: getRandomString(),
+				widgetName:
+					'com_liferay_site_navigation_language_web_portlet_SiteNavigationLanguagePortlet',
+			});
+
+			const layout = await apiHelpers.headlessDelivery.createSitePage({
+				pageDefinition: getPageDefinition([widgetDefinition]),
+				siteId: site.id,
+				title: getRandomString(),
+			});
+
+			await page.goto(
+				`/web${site.friendlyUrlPath}${layout.friendlyUrlPath}`
+			);
+		});
+
+		await test.step('Click on the Language Selector widget', async () => {
+			await clickAndExpectToBeVisible({
+				target: page.getByRole('menu'),
+				trigger: page.getByRole('button', {
+					name: 'Select a Language',
+				}),
+			});
+
+			const flags = EXPECTED_FLAG_CODES.map((code) =>
+				page.locator(`svg.lexicon-icon.lexicon-icon-${code}`)
```

Comment:
We should be more specific targeting icons. We need to check for `en-us` on the button, then the rest of the languages in the menu. We need to locate the menu via `aria-labelledby`. It should look something like this:

```
const trigger = page.getByRole('button', {
	name: 'Select a Language',
});

const triggerId = await trigger.getAttribute('id');

await clickAndExpectToBeVisible({
	target: page.getByRole('menu'),
	trigger: trigger,
});

const flags = EXPECTED_FLAG_CODES.map((code) =>
	page.locator(`[aria-labelledby="${triggerId}"] .lexicon-icon-${code}`)
);
```